$OpenBSD: patch-src_imlib_c,v 1.13 2012/03/29 12:29:10 dcoppa Exp $

fix memory leak when encountering unloadable file
(upstream git commit 8c8c1d5d52096a24e751f07d94972253a0e5054c)

--- src/imlib.c.orig	Thu Mar 29 14:19:43 2012
+++ src/imlib.c	Thu Mar 29 14:20:19 2012
@@ -305,6 +305,7 @@ static char *feh_magick_load_image(char *filename)
 		if (!WIFEXITED(status) || (WEXITSTATUS(status) != 0)) {
 			close(fd);
 			unlink(sfn);
+			free(sfn);
 			sfn = NULL;
 
 			if (!opt.quiet) {
